libraries/AP_AHRS/AP_AHRS_NavEKF.cpp

```diff
@@ -1475,3 +1478,65 @@ uint8_t AP_AHRS_NavEKF::get_primary_gyro_index(void) const
 
 #endif // AP_AHRS_NAVEKF_AVAILABLE
 
+/*
+ * Update AOA and SSA estimation based on airspeed, velocity vector and wind vector
+ *
+ * Based on:
+ * "On estimation of wind velocity, angle-of-attack and sideslip angle of small UAVs using standard sensors" by
+ * Tor A. Johansen, Andrea Cristofaro, Kim Sorensen, Jakob M. Hansen, Thor I. Fossen
+ *
+ * "Multi-Stage Fusion Algorithm for Estimation of Aerodynamic Angles in Mini Aerial Vehicle" by
+ * C.Ramprasadh and Hemendra Arya
+ *
+ * "ANGLE OF ATTACK AND SIDESLIP ESTIMATION USING AN INERTIAL REFERENCE PLATFORM" by
+ * JOSEPH E. ZEIS, JR., CAPTAIN, USAF
+ */
+void AP_AHRS_NavEKF::update_AOA_SSA(void)
+{
+	Quaternion aoa_NED2BODY;
+	Vector3f aoa_velocity, aoa_wind;
+
+	// get required data
+	if (get_velocity_NED(aoa_velocity) == false) {
+		return;
+	}
+	aoa_wind = wind_estimate();
+
+	// depending on EKF type choose right NED to Body Quaternion or matrix
+	switch (ekf_type()) {
```

Comment:
why have the EKF type specific code here? We can just use the generic get_rotation_body_to_ned() I think
We could also add a get_rotation_ned_to_body() to make it clearer